File: AtCoder/Greens/cpp/ARC040-B-Chokusen_Nuri.cpp
#include <bits/stdc++.h>
#define MAX_N 100
using namespace std;
typedef long long ll;

ll N, R;
string S;

int main() {
  cin >> N >> R >> S;
  ll last = N - 1;
  while (last >= 0 && S[last] == 'o')
    --last;
  ll i = 0;
  ll res = max(0LL, last - R + 1);
  while (i <= last) {
    if (S[i] == 'o') {
      ++i;
    } else {
      ++res;
      i += R;
    }
  }
  printf("%lld\n", res);
}

File: AtCoder/Greens/python/ARC040-B-Chokusen_Nuri.py
N, R = map(int, input().split())
S = input()
last = N - 1
while last >= 0 and S[last] == 'o':
    last -= 1
res = max(0, last - R + 1)
i = 0
while i <= last:
    if S[i] == 'o':
        i += 1
    else:
        res += 1
        i += R
print(res)
